webflow-hacks/cms-hacks/58/58.ts

```diff
@@ -0,0 +1,66 @@
+// inject the youtube api script
+const tag = document.createElement('script');
+tag.src = 'https://www.youtube.com/iframe_api';
+document.head.insertAdjacentElement('afterbegin', tag);
+
+/**
+ * default function called by the youtube api
+ */
+function onYouTubeIframeAPIReady() {
+  const IFRAME_SELECTOR = '[fs-hacks-element="iframe"]';
+  const iFrameContainer = document.querySelector(IFRAME_SELECTOR);
+  if (!iFrameContainer) return;
+
+  const iFrames = iFrameContainer.querySelectorAll('iframe');
+  iFrames.forEach((iFrame, index) => {
+    // set iframe properties
+    const { src } = iFrame;
+    if (!src) return;
+
+    const newSrc = new URL(src);
+    if (!newSrc.origin.includes('youtube')) return;
+
+    newSrc.searchParams.append('enablejsapi', '1');
+
+    iFrame.src = newSrc.toString();
+    iFrame.id = 'dynamic' + index;
+
+    createPlayer(iFrame);
+  });
+}
+
+/**
+ * Set up the player, add listeners, and play the video.
+ * @param iframe The iframe
+ */
+function createPlayer(iframe: HTMLIFrameElement) {
+  // initialize YT.player with the specified iframe's id
+  const player = new YT.Player(iframe, {
+    events: {
+      onReady: onPlayerReady,
+    },
+  });
+
+  function onPlayerReady() {
+    const timestampLinks = document.querySelectorAll<HTMLAnchorElement>('.hacks-rich-text a');
+    timestampLinks.forEach((timestampLink) => {
+      timestampLink.addEventListener('click', function (e) {
+        e.preventDefault();
+        const timestamp = timestampLink.innerText.trim();
+        const seconds = convertTimestampToSeconds(timestamp);
+        if (isNaN(seconds)) return;
+        player.seekTo(seconds, true);
+      });
+    });
+  }
+}
```

Comment:
You're parsing the timestamp every time the user clicks the button.
It would be better (performance-wise) if you use the closure to memoize the timestamp:
```typescript
    timestampLinks.forEach((timestampLink) => {
      const timestamp = timestampLink.innerText.trim();
      const seconds = convertTimestampToSeconds(timestamp);
      if (isNaN(seconds)) return;

      timestampLink.addEventListener('click', function (e) {
        e.preventDefault();
        player.seekTo(seconds, true);
      });
    });
  }
}
```

This way the calculation only happens one, and invalid timestamps don't even register a new event listener.